joomla/joomla-cms
# [4.0] Redo the fatal error pages

Pull Request as a helping hand for #30056 and  https://github.com/joomla/joomla-cms/pull/31735.

### Summary of Changes
- The tool is now creating 4 pages, incompatible, noxml, fatal and incomplete
- ~~The files are moved to `media/system` as these are static files~~
- ~~Fixed all the paths for this change~~

### Testing Instructions
Run `npm install` or `node build/build.js --build-pages` from the root path of your Joomla cloned repo
Check `/templates/system` for 4 html files: incompatible, noxml, fatal-error and build_incomplete
Open each in your browser (double click also works here)
You should have these:
![Screenshot 2020-12-20 at 21 17 57](https://user-images.githubusercontent.com/3889375/102723540-ef636500-4308-11eb-958f-3054b8526543.png)
![Screenshot 2020-12-20 at 21 18 07](https://user-images.githubusercontent.com/3889375/102723544-f2f6ec00-4308-11eb-8da2-10992339205c.png)
![Screenshot 2020-12-20 at 21 18 17](https://user-images.githubusercontent.com/3889375/102723546-f4281900-4308-11eb-86a3-4fbe648001af.png)
![Screenshot 2020-12-20 at 21 18 26](https://user-images.githubusercontent.com/3889375/102723548-f5594600-4308-11eb-8857-b92fb5b3dbf6.png)


Testing the actual php requires some editing of files (that's because it needs to kick the error):
- Front end change `<` to `>` https://github.com/joomla/joomla-cms/blob/f637260244cea4f81259c3bfdbda612c60b6090b/index.php#L18 and check your front end (eg localhost/)
- Back end change `<` to `>` https://github.com/joomla/joomla-cms/blob/f637260244cea4f81259c3bfdbda612c60b6090b/administrator/index.php#L18 and check your front end (eg localhost/)
- Installation change `<` to `>` https://github.com/joomla/joomla-cms/blob/f637260244cea4f81259c3bfdbda612c60b6090b/installation/index.php#L18 also rename your root configuration.php to configuration.back and check your front end (eg localhost/)
- Installation change `/media/vendor` to `/media/notexisting` https://github.com/joomla/joomla-cms/blob/f637260244cea4f81259c3bfdbda612c60b6090b/installation/includes/app.php#L18 also rename your root configuration.php to configuration.back and check your front end (eg localhost/)

There are also 2 more cases that require this PR to be merge, then pull #30056 and 31735 and follow the testing instructions on these PR's

That's all.
